frontend/packages/shared/src/components/GiteaHeader/VersionControlButtons/components/FetchChangesPopover/FetchChangesPopover.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { useState } from 'react';
+import { StudioPopover } from '@studio/components';
+import { DownloadIcon } from '@studio/icons';
+import classes from './FetchChangesPopover.module.css';
+import { useTranslation } from 'react-i18next';
+import { Notification } from '../Notification';
+import { useRepoPullQuery } from 'app-shared/hooks/queries';
+import { useStudioEnvironmentParams } from 'app-shared/hooks/useStudioEnvironmentParams';
+import { useQueryClient } from '@tanstack/react-query';
+import { GiteaFetchCompleted } from '../GiteaFetchCompleted';
+import { useVersionControlButtonsContext } from '../../context';
+import { FetchingFromGitea } from '../FetchingFromGitea';
+
+export const FetchChangesPopover = (): React.ReactElement => {
+  const {
+    isLoading,
+    setIsLoading,
+    hasMergeConflict,
+    commitAndPushChanges,
+    repoStatus,
+    onPullSuccess,
+  } = useVersionControlButtonsContext();
+
+  const { t } = useTranslation();
+  const { org, app } = useStudioEnvironmentParams();
+  const { refetch: fetchPullData } = useRepoPullQuery(org, app, true);
+  const queryClient = useQueryClient();
+
+  const [popoverOpen, setPopoverOpen] = useState(false);
+
+  const displayNotification: boolean = (repoStatus?.behindBy > 0 ?? false) && !hasMergeConflict;
+
+  const handleClosePopover = () => setPopoverOpen(false);
+
+  const handleOpenPopover = async () => {
+    setPopoverOpen(true);
+
+    setIsLoading(true);
+    const { data: result } = await fetchPullData();
+    setIsLoading(false);
+
+    if (result.repositoryStatus === 'Ok') {
+      if (onPullSuccess) onPullSuccess();
+
+      await queryClient.invalidateQueries(); // Todo: This invalidates ALL queries. Consider providing a list of relevant queries only.
+    } else if (result.hasMergeConflict || result.repositoryStatus === 'CheckoutConflict') {
+      await commitAndPushChanges('');
```

Comment:
Did we get an empty commit message here? What about adding a default message? 🤔

Reply:
What happens if the commitAndPushChanges promise is rejected? 😊 Would it be easier to use the useRepoCommitAndPushMutation directly here? That way, we could handle the error directly from the TanStack Query result. 😊

Reply:
This was just moved from the old code without making any considerations to it as it was not too relevant for this current issue 😄 
We can do a call tomorrow to see the possibilities